.github/workflows/code-quality.yml

```diff
@@ -0,0 +1,54 @@
+---
+name: Code Quality
+
+on:
+  push:
+    branches: ["master"]
+  pull_request:
+    branches: ["master", "4-cicd-init"]
+
+jobs:
+  lint:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+      - name: Install flake8
+        run: |
+          python -m pip install --upgrade pip==24.0
+          python -m pip install flake8==7.0.0 flake8-bugbear==24.4.21
```

Comment:
Prefer listing these development dependencies as optional dependencies in the `setup.py` file. That way there is a single source of truth for the package versions. You can then install the dependencies with the syntax for installing optional dependencies e.g. `pip install .[dev]`.